src/routing_table/mod.rs

```diff
@@ -909,7 +909,11 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
         self.they_want_to_merge = false;
         self.merge(&merge_prefix);
         self.add_missing_prefixes();
-        let targets = self.sections.keys().cloned().collect();
+        let targets = self.sections
```

Comment:
Include a comment here explaining why this happens please (copy from PR/commit description if need be).

Otherwise looks good.